### PR TITLE
Update how to enable Swift Package Manager instructions

### PR DESCRIPTION
**Pull request by loic-sharma**
> [!WARNING]
> Don't land this until https://github.com/flutter/flutter/pull/152049 has passed post-submit tests. That PR is likely to be reverted due to post-submit failures.

https://github.com/flutter/flutter/pull/152049 enables SwiftPM by default on the `main` channel. This updates the docs accordingly.

I also moved the "how to turn on/off SPM" guides down. This makes it so that the "How to migrate app" is the first section on the app developers page, and that the "How to migrate plugin" is the first section on the plugin developers page. This was feedback from today's Swift Package Manager walkthrough meeting.

Part of https://github.com/flutter/flutter/issues/151567

_PRs or commits this PR depends on (if any):_ https://github.com/flutter/flutter/pull/152049

## Presubmit checklist

- [x] This PR is marked as draft with an explanation if not meant to land until a future stable release.
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by sfshaza2 — APPROVED** [review]
lgtm, when ready

**Comment by loic-sharma** [comment]
https://github.com/flutter/flutter/pull/152049 got reverted. I'm investigating the fix and will update this thread once fix lands!

**Comment by loic-sharma** [comment]
Oof the fixes look non-trivial. I'll close this for now and re-open when we're ready to try this again. Apologies for the noise!

**Pull request closed without merging**
